pftools/python/parflow/tools/tests/pfb_peek.py

```diff
@@ -0,0 +1,169 @@
+"""
+    File: pfb_peek.py
```

Comment:
I agree here generally, although to be honest I don't see the need for the PFBPeek/PFBSummary at all - in my reading it basically just re-implements some base functionality that the ParflowBinaryReader class already has, and has some uninuitive behaviors like having methods which assume `argv` which is almost never something reasonable to assume in python. I think this was supposed to be a CLI that is akin `ncdump` for `pfb` files, but I'm not sure putting it in the tests directory really makes sense in that case either. @wh3248 - what are your thoughts?

Anyhow, for now I have renamed this to `PFBSummary`. Having a class instantiation return something is also not very common in python so instead of allowing `summary = PFBSummary()` I've simply implemented `pfb_summary(filename)` which will print out the summary.

Reply:
I agree the pattern for this utility doesn't look good.   I would expect the arg parsing to be in a main like free method that then constructs the summary object.   This doesn't seem to 'separate concerns' very well.

I'm not sure why the subgrid output was restricted to first 2 and last, or maybe have a flag to print all vs a few?   If one is looking at it for debugging, the hash will tell you something is wrong but if a middle subgrid is wrong we get no info on what/which one.

But since it was a stand-alone little script I was trying not to get super picky, better is the enemy of done.

Reply:
> But since it was a stand-alone little script I was trying not to get super picky, better is the enemy of done.

Totally agree :) 

I do think it would be pretty cool to make a little CLI script that fleshes this out at some point, but for another day.